src/transpiler.js

```diff
@@ -23,41 +23,63 @@ const transpiler = {
 module.exports = transpiler
 
 function getAddress({ addresses = [] }) {
-  return addresses.map(address => ({
-    city: address.city,
-    country: address.country,
-    postcode: address.postalCode,
-    street: address.streetAddress,
-    primary: address.metadata.primary || false
-  }))
+  return addresses.map(
+    address =>
+      address && {
+        city: address.city,
+        country: address.country,
+        postcode: address.postalCode,
+        street: address.streetAddress,
+        primary: address.metadata.primary || false,
+        type: address.type,
+        label: address.formattedType
+      }
+  )
 }
 
 function getBirthday({ birthdays = [] }) {
   return birthdays
-    .filter(birthday => birthday.metadata.primary)
-    .map(birthday => birthday.text)[0]
+    .filter(
+      birthday => birthday && birthday.metadata && birthday.metadata.primary
+    )
+    .map(
+      birthday =>
+        birthday &&
+        birthday.date &&
+        /\d\d\d\d-\d\d-\d\d/.exec(
+          new Date(Object.values(birthday.date).join('-')).toISOString()
+        )[0]
+    )[0]
 }
 
 function getEmail({ emailAddresses = [] }) {
-  return emailAddresses.map(email => ({
-    address: email.value,
-    primary: email.metadata.primary || false
-  }))
+  return emailAddresses.map(
+    email =>
+      email && {
+        address: email.value,
+        primary: email.metadata.primary || false,
+        type: email.type,
+        label: email.formattedType
+      }
+  )
 }
 
 function getPhone({ phoneNumbers = [] }) {
-  return phoneNumbers.map(phone => ({
-    number: phone.value,
-    primary: phone.metadata.primary || false
-  }))
+  return phoneNumbers.map(
+    phone =>
+      phone && {
+        number: phone.value,
+        primary: phone.metadata.primary || false,
+        type: phone.type,
+        label: phone.formattedType
+      }
+  )
 }
 
 function getGivenName({ names = undefined }) {
-  return names ? names[0].givenName : ''
+  return names && names[0] && names[0].givenName
```

Comment:
We don't want a fallback value anymore ?

Reply:
No, we stay as close as possible from the source's values.